cppcache/src/ExceptionTypes.cpp

```diff
@@ -39,294 +40,338 @@ void GfErrTypeThrowException(const char* str, GfErrType err) {
   }
   switch (err) {
     case GF_NOTCON: {
-      NotConnectedException ex(
-          str, (exMsg != nullptr ? exMsg : ": not connected to Geode"));
+      std::stringstream ss;
+      ss << str << (exMsg != nullptr ? exMsg : ": not connected to Geode");
+      NotConnectedException ex(ss.str());
```

Comment:
Nitpick here, but I think you can use a string and append the values you want here. I think it would be faster, though  not by a whole lot. Why go that way?

Reply:
Nitpick asside, I wonder if some duplicate code could be removed but moving some things:
```
std::stringstream ss;
ss << str;
switch ... {
  case GF_NOTCON
     ss << (exMsg ? exMsg : ": not connected to Geode")
...
```

Reply:
Sounds good. I incorporated in next change set.